src/Elastic.Apm.Specification/Validator.cs

```diff
@@ -0,0 +1,657 @@
+						else
+						{
+							result.AddError(
+								TypeValidationError.ExpectedType(
+									specType,
+									specTypeProperty.PropertyType,
+									"string",
+									schema.GetNameOrSpecificationId(),
+									name));
+						}
+						break;
+					case "System.Boolean":
+						if (schemaProperty.Type.HasFlag(JsonObjectType.Boolean))
+							ValidateBoolean(specType, schema, schemaProperty, specTypeProperty, result);
+						else
+						{
+							result.AddError(
+								TypeValidationError.ExpectedType(
+									specType,
+									specTypeProperty.PropertyType,
+									"boolean",
+									schema.GetNameOrSpecificationId(),
+									name));
+						}
+						break;
+					default:
+						// Are there multiple types? If so, based on the .NET type not being a primitive type, we would expect the presence of the
+						// schema "object" or "array" type in the majority of cases, so check these first.
+						// For types with custom serialization, the schema type may be a primitive type, so we can't easily statically validate it.
+						if (HasMultipleNonNullTypes(schemaProperty.Type))
+						{
+							if (schemaProperty.Type.HasFlag(JsonObjectType.Object))
+								ValidateProperties(specTypeProperty.PropertyType, schemaProperty, result);
+							else if (schemaProperty.Type.HasFlag(JsonObjectType.Array) &&
+								IsEnumerableType(specTypeProperty.PropertyType, out var elementType))
+								ValidateProperties(elementType, schemaProperty.Item.ActualSchema, result);
+							else
+							{
+								result.AddIgnore(new TypeValidationIgnore(schema.GetNameOrSpecificationId(), name,
+									$"Cannot statically check type. .NET type '{specType}', schema type '{schemaProperty.Type}'"));
+							}
+						}
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.Boolean))
+							ValidateBoolean(specType, schema, schemaProperty, specTypeProperty, result);
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.Integer))
+							ValidateInteger(specType, schema, schemaProperty, specTypeProperty, result);
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.Number))
+							ValidateNumber(specType, schema, schemaProperty, specTypeProperty, result);
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.String))
+						{
+							if (schemaProperty.IsEnumeration && specTypeProperty.PropertyType.IsEnum)
+								ValidateEnum(specType, schema, schemaProperty, specTypeProperty, result);
+							else
+								ValidateString(specType, schema, schemaProperty, specTypeProperty, result);
+						}
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.Object))
+							ValidateProperties(specTypeProperty.PropertyType, schemaProperty, result);
+						else if (schemaProperty.Type.HasFlag(JsonObjectType.Array))
+						{
+							if (IsEnumerableType(specTypeProperty.PropertyType, out var elementType))
+								ValidateProperties(elementType, schemaProperty.Item.ActualSchema, result);

```

Comment:
In case of `elementType=Elastic.Apm.Api.CapturedStackFrame` `schemaProperty.Item`  is `null`. This happens on the `Error` validation.

Reply:
This was a bug in the spec and has been fixed in https://github.com/elastic/apm-server/issues/4515